src/Google/Cache/File.php

```diff
@@ -151,11 +151,14 @@ private function getCacheDir($file, $forWrite)
     // use the first 2 characters of the hash as a directory prefix
     // this should prevent slowdowns due to huge directory listings
     // and thus give some basic amount of scalability
-    $dirHash = substr(md5($file), 0, 2);
+    $fileHash = substr(md5($file), 0, 2);
+    $userHash = md5(get_current_user());
+    $dirHash = $fileHash . DIRECTORY_SEPARATOR . $userHash;
```

Comment:
Can you change the order of them?
```php
$dirHash = $userHash . DIRECTORY_SEPARATOR . $fileHash;
```
Then there will be no permission issue.

Also, maybe you can just use the username, instead of a hash of the username? What's the purpose of hashing the username? The md5 hash collision may be unlikely, but from the security standpoint, it needs to be theoretically perfect.